### ISSUE TITLE
 IMS call One way Voice and call disconnect in 3 second not see any PDU modification request with QCI 1 

### ISSUE THREAD

**Issue by sonutelcoandai** [herlesupreeth/kamailio#3]
Issue :  IMS call One way Voice and call disconnect in 3 second not see any PDU modification request with QCI 1 

Important info:-  UE:- Samsung S21, Oneplus9
                           GnodeB:- Sercom
                           Kamalio and Open5GS installed.
                          Call Connect and UE registration Normal.
                         All setup are on Same VM kamalio and Open5GS.

Issue reproduce steps:--

Register both UE, Call from Oneplus UE to Samsung>>> than  Call is connected>> Receive at Samsung UE>>> Observe the Issue one way sound oneplus have voice but Samsung no voice>>> and call got disconnected.

Expected behavior:-  Call connect successful and RTP packets flow normally. call should not disconnect with in 3 seconds.

download logs from Here :--   https://github.com/open5gs/open5gs/files/12122557/onewayvoiceissue.zip

![image](https://github.com/herlesupreeth/kamailio/assets/140157156/fe36ce15-47bf-4ae8-bedf-3d775921b741)
![image](https://github.com/herlesupreeth/kamailio/assets/140157156/b969398a-7370-45cc-b807-181071f7e429)

**Comment by herlesupreeth** [comment]
First of all kamailio P-CSCF does not implement SBI interface towards PCF (N5 interface) so its not possible to have VoNR working. Second, I see you have tried the above setup by disabling the Rx interface flag in kamailio cfg file. Rx interface is a diameter interface in 4G which allows P-CSCF to talk with PCRF in order to create dedicated bearers required for VoLTE call.

**Comment by sonutelcoandai** [comment]
Hi @herlesupreeth thanks for the reply.

can you confirm one thing, can we have a VoLte call, with current setup which we are using?
 Although we  have tried to enable RX interface in PCSCF_cfg but as you said its disabled, below is screenshot for your reference,

kindly let me know where we have to enable it ?

![image](https://github.com/herlesupreeth/kamailio/assets/140157156/60a115d8-c50b-4263-925c-c1dd7f205e03)

**Comment by herlesupreeth** [comment]
> can you confirm one thing, can we have a VoLte call, with current setup which we are using?

I dont think so. Since you are using a gNB (5G Radio), your gNB would be connected to SA part of the open5gs core network. And, open5gs does not support fallback mechanism from VoNR to VoLTE as far as I know.

If you want to test VoLTE with your setup you need to have eNB (4G Radio) to which UE is connected.

**Comment by sonutelcoandai** [comment]
thanks, very much.

is RX interface configuration are ok with above comment Screenshot?

is there any  IMS open source which currently supporting VoNR?

or are you planing  to implement SBI interface (N5) for VoNR call? or any way to configure this interface?

**Comment by likemkmishra** [comment]
Hi @herlesupreeth @sonu5gc,
I am also facing the same issue..
I have two UE (Oppo and oneplus) both UE is successfully registered on 5G network SA mode and open IMS(Kamailio) and I can see the VoLTE symbol on both UEs but voice is only coming on oneplus UE and call disconnected in 8secs.
I am also not able to find way to resolve this issue.
Hi @herlesupreeth Really appreciated your efforts to making this wonderful project..

**Comment by herlesupreeth** [comment]
@sonu5gc 

> is RX interface configuration are ok with above comment Screenshot?

I believe by default RX interface is enabled and are enabled/disabled in pcscf.cfg file.

> is there any IMS open source which currently supporting VoNR?

Sadly none.

> or are you planing to implement SBI interface (N5) for VoNR call? or any way to configure this interface?

No. I stopped working on IMS and currently working on gNB. Its not possible to configure using cfg file rather you would have to put your hands into kamailio C code.

@likemkmishra 

> I have two UE (Oppo and oneplus) both UE is successfully registered on 5G network SA mode and open IMS(Kamailio)

I would not consider the IMS registration as successful unless P-CSCF is able to talk with PCF. From the perspective of SIP signalling, yes, its correct and registration is successful. But that's just one part of the puzzle. And, in order for VoNR call to work, dedicated bearers are needed, which are created when P-CSCF talks with PCF over N5 interface.

**Comment by sonutelcoandai** [comment]
rather you would have to put your hands into kamailio C code.

Can you give any idea about this, like which entity code we have to change or modify if i try ?

what things must need to modify in code.

Much appreciated.

**Comment by herlesupreeth** [comment]
Its most of the files under the following module

> https://github.com/herlesupreeth/kamailio/tree/5.3/src/modules/ims_qos

> what things must need to modify in code.

This I cannot tell as I have not read the specification of N5. But if I were to approach the implementation I would start by adding feature to talk HTTP2 to NRF/SCP and send register with it and advertise the NF services. Once this is complete I would move on to next step of implementing the required NF service of N5 interface.
